File: pascals_triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = [[1]]

        for _ in range(numRows - 1):
            dummy_row = [0] + res[-1] + [0]
            row = []

            for i in range(len(res[-1]) + 1):
                row.append(dummy_row[i] + dummy_row[i+1])
            res.append(row)
        
        return res

File: pascals_triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        std::vector<std::vector<int>> res;
        res.push_back({1});

        for (int i = 0; i < numRows - 1; i++) {
            std::vector<int> dummyRow = {0};
            dummyRow.insert(dummyRow.end(), res.back().begin(), res.back().end());
            dummyRow.push_back(0);
            std::vector<int> row;

            for (int j = 0; j < dummyRow.size() - 1; j++) {
                row.push_back(dummyRow[j] + dummyRow[j + 1]);
            }

            res.push_back(row);
        }

        return res;        
    }
};
